init.js

```diff
@@ -96,82 +185,29 @@ function loadData(modName, settings) {
         solver = new Solver(items, recipes)
 
         renderSettings(settings)
-
-        if ("items" in settings && settings.items != "") {
-            var targets = settings.items.split(",")
-            for (var i=0; i < targets.length; i++) {
-                var targetString = targets[i]
-                var parts = targetString.split(":")
-                var name = parts[0]
-                var target = addTarget(name)
-                var type = parts[1]
-                if (type == "f") {
-                    target.setFactories(parts[2])
-                } else if (type == "r") {
-                    target.setRate(parts[2])
-                } else {
-                    throw new Error("unknown target type")
-                }
-            }
-        } else {
-            addTarget()
-        }
-        if ("modules" in settings && settings.modules != "") {
-            var moduleSettings = settings.modules.split(",")
-            for (var i=0; i < moduleSettings.length; i++) {
-                var bothSettings = moduleSettings[i].split(";")
-                var factoryModuleSettings = bothSettings[0]
-                var beaconSettings = bothSettings[1]
-
-                var singleModuleSettings = factoryModuleSettings.split(":")
-                var recipeName = singleModuleSettings[0]
-                var recipe = recipes[recipeName]
-                var factory = spec.getFactory(recipe)
-                var moduleNameList = singleModuleSettings.slice(1)
-                for (var j=0; j < moduleNameList.length; j++) {
-                    var moduleName = moduleNameList[j]
-                    if (moduleName && moduleName != "null") {
-                        var module
-                        if (moduleName in modules) {
-                            module = modules[moduleName]
-                        } else if (moduleName in shortModules) {
-                            module = shortModules[moduleName]
-                        }
-                        if (module) {
-                            factory.setModule(j, module)
-                        }
-                    }
-                }
-                if (beaconSettings) {
-                    beaconSettings = beaconSettings.split(":")
-                    var moduleName = beaconSettings[0]
-                    var module
-                    if (moduleName in modules) {
-                        module = modules[moduleName]
-                    } else if (moduleName in shortModules) {
-                        module = shortModules[moduleName]
-                    }
-                    if (module) {
-                        var count = RationalFromFloat(Number(beaconSettings[1]))
-                        factory.beaconModule = module
-                        factory.beaconCount = count
-                    }
-                }
-            }
-        }
-        initDone = true
-        itemUpdate()
+        loadModules(settings)
+        loadItems(settings)
```

Comment:
Why did you change the order of loading the build targets and loading the module spec?